dev/bots/utils.dart

```diff
@@ -569,9 +570,26 @@ List<T> selectIndexOfTotalSubshard<T>(List<T> tests, {String subshardKey = kSubs
     return <T>[];
   }
 
-  final int testsPerShard = (tests.length / total).ceil();
-  final int start = (index - 1) * testsPerShard;
-  final int end = math.min(index * testsPerShard, tests.length);
+  // While there exists a closed formula figuring out the range of tests this
```

Comment:
Can we unit test this code, given it had a bug and how hard it was to fix it?

Reply:
I would recommend pulling this into its own function, to make testing easier--when doing that, I would recommend passing the index as `index - 1`, so that the function can just pretend its 0-indexed.

Reply:
> Can we unit test this code, given it had a bug and how hard it was to fix it?

Done

> I would recommend passing the index as `index - 1`, so that the function can just pretend its 0-indexed.

I would like any parameter representing the subshard index to represent it exactly. For example, if the current subshard is `1_9`, then I would be (slightly) surprised if a variable named `subshardIndex` (or similar) was set to `0`.